rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,291 @@
+            return False
+
+        assert action_items[-1].extra_data is not None
+        asset = event.asset.resolve_to_evm_token()
+        to_amount = asset_normalized_value(
+            amount=action_items[-1].extra_data['amount_out'],
+            asset=asset,
+        )
+
+        if (
+            action_items[-1].extra_data['to_token'] != tx_log.address or
+            event.balance.amount != to_amount
+        ):
+            return False
+
+        is_receive = asset.evm_address == action_items[-1].extra_data['to_token']
+
+        event.counterparty = CPT_BALANCER_V2
+        event.event_type = HistoryEventType.TRADE
+        if is_receive:
+            event.event_subtype = HistoryEventSubType.RECEIVE
+            event.notes = f'Receive {event.balance.amount} {asset.symbol} in balancer V2 from {event.location_label}'  # noqa: E501
+        else:
+            event.event_subtype = HistoryEventSubType.SPEND
+
+        return True
+
+    def _maybe_enrich_balancer_v1_events(
+        self,
+        token: 'EvmToken',
+        tx_log: EvmTxReceiptLog,  # pylint: disable=unused-argument
+        transaction: EvmTransaction,  # pylint: disable=unused-argument
+        event: HistoryBaseEntry,
+        action_items: list[ActionItem],  # pylint: disable=unused-argument
+        all_logs: list[EvmTxReceiptLog],
+    ) -> bool:
+        actions = self._decode_v1_pool_event(all_logs=all_logs)
+        if actions is None:
+            return False
+
+        for action in actions:
+            assert action.extra_data is not None
+            if (
+                action.extra_data['type'] == BalancerV1EventTypes.JOIN and
+                action.extra_data['ds_address'] == event.counterparty
+            ):
+                if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    # otherwise this is the event that we have to edit
+                    event.event_type = HistoryEventType.STAKING
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Receive {event.balance.amount} {token.symbol} as deposit in Balancer V1 pool'  # noqa: E501
+                    return True
+                if (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    # otherwise this is the event that we have to edit
+                    event.event_type = HistoryEventType.STAKING
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Deposit {event.balance.amount} {token.symbol} in Balancer V1 pool'  # noqa: E501
+                    return True
+            if (
```

Comment:
This should have been elif no?
```suggestion
            elif (
```

Reply:
No since all `if` conditions above have a return statement

Reply:
You are misreading the code, which shows why such ifs are bad. The ifs that have a return statement are indented. There is only one if above this, and it does not end with a return.